File: ios/RNSignInWithApple/RNSingInButtonViewManager.m
//
//  RNSignInButtonViewManager.m
//  SignInWithApple
//
//  Created by Andrej on 19/06/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(SingInButtonViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(buttonTypeRawValue, NSInteger)
RCT_EXPORT_VIEW_PROPERTY(buttonStyleRawValue, NSInteger)
RCT_EXPORT_VIEW_PROPERTY(cornerRadius, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(onPress, RCTBubblingEventBlock)

@end

File: ios/lib/RNSingInButtonViewManager.swift
//
//  SignInButtonViewManager.swift
//  SignInWithApple
//
//  Created by Andrej on 17/06/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation
import AuthenticationServices

@available(iOS 13.0, *)
@objc(SingInButtonViewManager)
class SingInButtonViewManager: RCTViewManager {
  
  var innerView: AppleIDButtonView?;
 
  @objc override static func requiresMainQueueSetup() -> Bool {
    return false;
  }
  
  @objc override func view() -> AppleIDButtonView? {
    innerView = AppleIDButtonView();
    innerView?.appleIDButtonDelegate = self;
    return innerView;
  }

}

@available(iOS 13.0, *)
extension SingInButtonViewManager: AppleIDButtonDelegate {

  func buttonWasPressed(button: ASAuthorizationAppleIDButton) -> Void {
    self.innerView!.onPress!(nil);
  }
  
}
